StratisCore.UI/src/app/wallet/smart-contracts/components/address-selection/address-selection.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit } from '@angular/core';
+import { GlobalService } from '@shared/services/global.service';
+import { SmartContractsServiceBase } from '../../smart-contracts.service';
+import { ClipboardService } from 'ngx-clipboard';
+import { catchError, takeUntil, switchMap } from 'rxjs/operators';
+import { of, Subject } from 'rxjs';
+import { CurrentAccountService } from '@shared/services/current-account.service';
+import { Router } from '@angular/router';
+import { ApiService } from '@shared/services/api.service';
+import { WalletInfo } from '@shared/models/wallet-info';
+import { WalletService } from "@shared/services/wallet.service";
+
+@Component({
+  selector: 'app-address-selection',
+  templateUrl: './address-selection.component.html',
+  styleUrls: ['./address-selection.component.css']
+})
+export class AddressSelectionComponent implements OnInit {
+
+  private walletName = '';
+  addresses: any[];
+  addressChangedSubject: Subject<string>;
+  selectedAddress: any;
+  coinUnit: string;
+  unsubscribe: Subject<void> = new Subject();
+
+  constructor(private globalService: GlobalService,
+              private walletService: WalletService,
+              private currentAccountService: CurrentAccountService,
+              private router: Router,
+              private clipboardService: ClipboardService) {
+
+    this.coinUnit = this.globalService.getCoinUnit();
+    this.walletName = this.globalService.getWalletName();
+    this.addressChangedSubject = new Subject();
+
+    this.walletService
+      .getAllAddressesForWallet(new WalletInfo(this.walletName))
+      .pipe(
+        catchError(error => {
+          console.log('Error retrieving addressses. ' + error);
```

Comment:
use logger service and maybe pass error as a second parameter so that you get a stack trace `Log.error('Error retrieving addressses. ', error);`